includes/Core/Util/Reset.php

```diff
@@ -67,10 +67,48 @@ function( $routes ) {
 	 * @since 1.0.0
 	 */
 	public function all() {
-		$googlesitekit_reset_context = $this->context;
+		global $wpdb;
 
-		// Call uninstaller.
-		require $this->context->path( 'uninstall.php' );
+		$sitekit_key_pattern = 'googlesitekit\_%';
+
+		if ( $this->context->is_network_mode() ) {
+			$meta_key_pattern = $sitekit_key_pattern;
+			$options_table    = $wpdb->sitemeta;
+			$options_column   = 'meta_key';
+			$transient_prefix = '_site_transient_';
+		} else {
+			$meta_key_pattern = $wpdb->get_blog_prefix() . $sitekit_key_pattern;
+			$options_table    = $wpdb->options;
+			$options_column   = 'option_name';
+			$transient_prefix = '_transient_';
+		}
+
+		// Delete options and transients.
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		$wpdb->query(
+			$wpdb->prepare(
+				/* phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared */
+				"
+				DELETE FROM $options_table
+				WHERE  $options_column LIKE %s
+					OR $options_column LIKE %s
+					OR $options_column LIKE %s
+					OR $options_column = %s
+				", /* phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared */
+				$sitekit_key_pattern,
+				$transient_prefix . $sitekit_key_pattern,
+				$transient_prefix . 'timeout_' . $sitekit_key_pattern,
+				'googlesitekit-active-modules'
+			)
+		);
```

Comment:
This changes the logic a bit. The original `uninstall.php` logic cleared the `$wpdb->options` table and the prefixed `usermeta` (user options) always. Only network tables are only cleared if network mode.

The new logic doesn't clear site data when the plugin is in network mode. While this seems more logical at first glance, deleting everything from (site) options as well was deliberately chosen before and should continue to work this way.

Potentially, we should even delete all data that _could_ exist, i.e. always delete prefixed `usermeta` and non-prefixed `usermeta`, and always delete network options as soon as `is_multisite()`. Not necessarily, but it's a consideration because this should really reset everything related to Site Kit.

Reply:
I've updated the method with two new internal helpers to make it very explicit what's happening now 👍 

Ready for another pass!